olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -363,13 +364,59 @@ bool DefaultCacheImpl::PromoteKeyLru(const std::string& key) {
   return true;
 }
 
+bool DefaultCacheImpl::PutMutableCache(const std::string& key,
+                                       const leveldb::Slice& value,
+                                       time_t expiry) {
+  if (!mutable_cache_) {
+    return true;
+  }
+
+  // can't put new item if cache is full and eviction disabled
+  const auto item_size = value.size();
+  const auto time_point =
+      std::to_string(InMemoryCache::DefaultTimeProvider()());
+  const auto expected_size = mutable_cache_data_size_ + item_size + key.size() +
+                             CreateExpiryKey(key).size() + time_point.size();
```

Comment:
Two things i dont like about this, but maybe lets consider this in a separate ticket:
- Creating a `std::to_string(InMemoryCache::DefaultTimeProvider()())` for no reason as most of the time the expiry is not even set; the size of the timestamp you can estimate (it will stay 10 digits long until 11/20/2286. I think this is enough precision for us.
- Creating a CreateExpiryKey(key) only to count it although here also you can estimate: `key.size() + strlen(kExpirySuffix)`, or if strlen is to risky for you a constexpr kExpirySuffixLength variable.